### PR TITLE
chore: Enable Network peering acc tests using AWS to run as part of CI

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Ticket: https://jira.mongodb.org/browse/CLOUDP-219839
This PR enables the network peering acc tests using AWS to run as part of our CI pipeline. It also updates the project ip access-list tests that were missed in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1784 to run in the CI pipeline

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### I run the test locally
Running tool: /Users/andrea.angiolillo/.asdf/shims/go test -timeout 300000s -run ^TestAccNetworkRSNetworkPeering_basicAWS$ github.com/mongodb/terraform-provider-mongodbatlas/internal/service/networkpeering

ok  	github.com/mongodb/terraform-provider-mongodbatlas/internal/service/networkpeering	177.808s

### I run the tests as part of the CI pipeline

- [TestAccNetworkRSPrivateLinkEndpointAWS_basic](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/7339055584/job/19983118342?pr=1793#step:5:326)
- [TestAccMigrationProjectDSProjectIPAccessList_SettingAWSSecurityGroup](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/7339055584/job/19983119425?pr=1793#step:5:319)
- [TestAccProjectDSProjectIPAccessList_SettingAWSSecurityGroup](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/7339055584/job/19983118189?pr=1793#step:5:381)

### CODE REVIEW

File: internal/service/projectipaccesslist/data_source_project_ip_access_list_migration_test.go [hunk 1]
```diff
@@ -83,38 +83,37 @@ func TestAccMigrationProjectDSProjectIPAccessList_SettingCIDRBlock(t *testing.T)
 }
 
 func TestAccMigrationProjectDSProjectIPAccessList_SettingAWSSecurityGroup(t *testing.T) {
-	acc.SkipTestExtCred(t)
+	projectName := acctest.RandomWithPrefix("test-acc-migration-project-aws")
```
Comment (andreaangiolillo) [inline internal/service/projectipaccesslist/data_source_project_ip_access_list_migration_test.go:86]: Follow up changes of https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1784 where I forgot to update the migration and data source tests

File: internal/service/networkpeering/resource_network_peering_test.go [hunk 2]
```diff
@@ -18,32 +18,45 @@ import (
 )
 
 func TestAccNetworkRSNetworkPeering_basicAWS(t *testing.T) {
```
Comment (AgustinBettati) [inline internal/service/networkpeering/resource_network_peering_test.go]:
```suggestion
func TestAccNetworkNetworkPeering_basicAWS(t *testing.T) {
```
Just to make explicit that this test includes both resource and data sources

**Review by AgustinBettati — APPROVED** [review]
LGTM, non blocking comment

**Review by maastha — APPROVED** [review]
LGTM

**Pull request merged**